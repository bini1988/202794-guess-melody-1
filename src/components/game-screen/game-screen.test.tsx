import React from "react";
import renderer from "react-test-renderer";
import GameScreen from "./game-screen";
import {TrackGenres, QuestionTypes, GameQuestion} from "../../types.d";

const questionsMock: GameQuestion[] = [
  {
    type: QuestionTypes.Genre,
    genre: TrackGenres.Rock,
    answers: [
      {src: `src#1`, genre: TrackGenres.Rock},
      {src: `src#2`, genre: TrackGenres.Jazz},
    ]
  }, {
    type: QuestionTypes.Artist,
    song: {artist: `artist#1`, src: `src#1`},
    answers: [
      {picture: `picture#1`, artist: `artist#1`},
      {picture: `picture#2`, artist: `artist#2`},
    ]
  }
];

it(`GameScreen correctly renders default markup`, () => {
  const handleAnswer = () => {};

  const tree = renderer
    .create(
        <GameScreen
          questions={questionsMock}
          questionIndex={0}
          time={5}
          mistakes={3}
          onAnswer={handleAnswer}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
