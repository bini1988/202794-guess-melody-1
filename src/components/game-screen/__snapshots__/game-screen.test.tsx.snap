// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScreen correctly renders default markup 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
    >
      <span
        className="timer__mins"
      >
        00
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        05
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите rock треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio>
            <source
              src="src#1"
              type="audio/ogg; codecs=vorbis"
            />
          </audio>
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-0"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="rock"
          />
          <label
            className="game__check"
            htmlFor="answer-0"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio>
            <source
              src="src#2"
              type="audio/ogg; codecs=vorbis"
            />
          </audio>
        </div>
        <div
          className="game__answer"
        >
          <input
            checked={false}
            className="game__input visually-hidden"
            id="answer-1"
            name="answer"
            onChange={[Function]}
            type="checkbox"
            value="jazz"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
